internal/cli/up.go

```diff
@@ -114,22 +121,34 @@ func (c *upCmd) initialTermMode(isTerminal bool) store.TerminalMode {
 }
 
 func (c *upCmd) run(ctx context.Context, args []string) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	a := analytics.Get(ctx)
+	defer a.Flush(time.Second)
 
-	termMode := c.initialTermMode(isatty.IsTerminal(os.Stdout.Fd()))
+	log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
+	isTTY := isatty.IsTerminal(os.Stdout.Fd())
+	termMode := c.initialTermMode(isTTY)
 
 	cmdUpTags := engineanalytics.CmdTags(map[string]string{
 		"update_mode": updateModeFlag, // before 7/8/20 this was just called "mode"
 		"term_mode":   strconv.Itoa(int(termMode)),
 	})
+
+	generateTiltfileResult, err := maybeGenerateTiltfile(c.fileName)
```

Comment:
Should we skip asking to generate a Tiltfile `if !isatty.IsTerminal(os.Stdin.Fd())`? If Tilt is run in the background without access to one, for example.

Reply:
Yep! There's a check at the beginning of the method - the `IsTerminal` check is cheap so it was easier to do redundantly to make the logic a bit more hermetic (and simplify the analytic logic)

https://github.com/tilt-dev/tilt/blob/ccb0908011bff82975193d7498ffde340cf5d6d4/internal/cli/generate_tiltfile.go#L33-L36